### PR TITLE
[Docs] Add consistent H5s to each example options

### PR DESCRIPTION
**Pull request by llorca**
Still a few rough edges:
screen shot 2018-05-16 at 3 20 18 pm
screen shot 2018-05-16 at 3 47 32 pm
screen shot 2018-05-16 at 3 48 27 pm
screen shot 2018-05-16 at 3 48 45 pm
screen shot 2018-05-16 at 3 52 54 pm

They're due to `core` spacing bugs. Good thing we're building a sidebar-style form options container, I've discovered a ton of form controls issues thanks to that!!

### CODE REVIEW

File: packages/docs-app/src/examples/core-examples/common/intentSelect.tsx [hunk 1]
```diff
@@ -16,12 +17,13 @@ const INTENTS = [
...
 
 export interface IIntentSelectProps {
+    inline?: boolean;
     intent: Intent;
     onChange: React.FormEventHandler<HTMLSelectElement>;
 }
 
 export const IntentSelect: React.SFC<IIntentSelectProps> = props => (
-    <label className={Classes.LABEL}>
+    <label className={classNames(Classes.LABEL, { [Classes.INLINE]: props.inline })}>
```
Comment (giladgray) [inline packages/docs-app/src/examples/core-examples/common/intentSelect.tsx]: use `Label` component which supports `inline` prop

File: packages/docs-app/src/examples/core-examples/tagInputExample.tsx [hunk 2]
```diff
@@ -92,11 +92,12 @@ export class TagInputExample extends React.PureComponent<IExampleProps, ITagInpu
     private renderOptions() {
         return (
             <>
+                <H5>Props</H5>
                 <Switch label="Fill container width" checked={this.state.fill} onChange={this.handleFillChange} />
                 <Switch label="Large" checked={this.state.large} onChange={this.handleLargeChange} />
                 <Switch label="Disabled" checked={this.state.disabled} onChange={this.handleDisabledChange} />
                 <Switch label="Add on blur" checked={this.state.addOnBlur} onChange={this.handleAddOnBlurChange} />
-                <H5>Tag props</H5>
+                <H5>Example</H5>
```
Comment (giladgray) [inline packages/docs-app/src/examples/core-examples/tagInputExample.tsx]: this should stay as "Tag props" -- these switches affect `tagProps` behavior, which is one of the props

File: packages/docs-app/src/examples/datetime-examples/dateRangePickerExample.tsx [hunk 3]
```diff
@@ -119,6 +120,7 @@ export class DateRangePickerExample extends React.PureComponent<IExampleProps, I
                     />
                 </div>
                 <div>
+                    <H5 />
```
Comment (giladgray) [inline packages/docs-app/src/examples/datetime-examples/dateRangePickerExample.tsx]: ??

File: packages/docs-app/src/examples/select-examples/selectExample.tsx [hunk 4]
```diff
@@ -82,16 +83,17 @@ export class SelectExample extends React.PureComponent<IExampleProps, ISelectExa
...
-                    label="Minimal popover style"
-                    checked={this.state.minimal}
-                    onChange={this.handleMinimalChange}
-                />
                 <Switch
                     label="Use initial content"
                     checked={this.state.hasInitialContent}
                     onChange={this.handleInitialContentChange}
                 />
+                <H5>Example</H5>
```
Comment (giladgray) [inline packages/docs-app/src/examples/select-examples/selectExample.tsx]: this is more like `Popover props`: it's very much a supported concept, not something contrived for the example

File: packages/docs-app/src/examples/datetime-examples/dateRangePickerExample.tsx (hunk 3 above)
Comment (llorca) [reply packages/docs-app/src/examples/datetime-examples/dateRangePickerExample.tsx]: it looks off without it, but yes it's gross so I'll remove it

**Pull request merged**